### ISSUE TITLE
Unable to load local file as $ref

### ISSUE THREAD

**Issue by ryanwalder** [chris48s/v8r#241]
I'm trying to modularize some schema definitions by splitting out common definitions into their own files but I'm having troube referencing them from the local file system. These are all local files, none from the json-schema site.

I feel like I'm missing something simple and after looking through the docs for ajv and v8r I'm still a bit lost!

Example:

`Dir layout`

```
schemas/components/price.json
schemas/ammo.json
src/packs/ammo/*.yaml
```

`.v8rrc.yaml`

```
customCatalog:
  schemas:
    - name: "Packs/Ammo"
      description: "Schema for validating Ammo YAML fragments"
      fileMatch: ["src/packs/ammo/*.yaml"]
      location: "schema/ammo.json"
```

`schemas/ammo.json`

Cut out quite a lot for brevity, hopefully still valid. It works with `price` as it's own inline definition (`"$ref": "#/definitions/Price"`), just not as a `$ref` to a file.

```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$ref": "#/definitions/CPRC-Ammo",
  "definitions": {
    "CPRC-Ammo": {
      "type": "object",
      "additionalProperties": false,
      "description": "Definition of an Ammo Compendium Item for FVTT system Cyberpunk RED - Core",
      "properties": {
        "system": {
          "description": "The system's values for the Ammo item",
          "$ref": "#/definitions/System"
        },
      "required": [
        "system"
      ],
      "title": "CPRC: Ammo Compendium Item"
    },
    "System": {
      "type": "object",
      "additionalProperties": false,
      "description": "CPRC Ammo Item definition",
      "properties": {
        "price": {
          "description": "Item Price,
          "$ref": "./components/price.json"
        },
      },
      "required": [
        "price"
      ],
      "title": "System"
    },
  }
}
```

`schemas/components/price.json`

```
{
  "type": "object",
  "additionalProperties": false,
  "properties": {
    "market": {
      "type": "integer"
    }
  },
  "required": [
    "market"
  ],
  "title": "Price"
}

```

`npx v8r src/packs/ammo/arrow.armor.piercing.yaml`

```
ℹ Loaded config file from .v8rrc.yaml
ℹ Processing src/packs/ammo/arrow.armor.piercing.yaml
ℹ Found schema in .v8rrc.yaml ...
ℹ Validating src/packs/ammo/arrow.armor.piercing.yaml against schema from /home/ryan/git/fvtt-cyberpunk-red-core/schema/ammo.json ...
✖ Failed fetching components/price.json

{
  "results": {
    "src/packs/ammo/arrow.armor.piercing.yaml": {
      "fileLocation": "src/packs/ammo/arrow.armor.piercing.yaml",
      "schemaLocation": "/home/ryan/git/fvtt-cyberpunk-red-core/schema/ammo.json",
      "valid": null,
      "errors": [],
      "code": 1
    }
  }
}
```

**Comment by chris48s** [comment]
Hi. I've just tried setting this up and can reproduce. My gut instinct is I probably need to replace the resolver (which is used to resolve external `$ref`s)

https://github.com/chris48s/v8r/blob/59e9424fe1f779bc75216dea46bf5699b60e3492/src/ajv.js#L13

with a call to `getFromUrlOrFile()`

https://github.com/chris48s/v8r/blob/59e9424fe1f779bc75216dea46bf5699b60e3492/src/io.js#L4-L8

but I've not tried it/written test-cases yet.

**Comment by chris48s** [comment]
This may actually be slightly more complicated. I need to be careful not to break resolution in the case where the schema is remote but contains relative references. Semi-related: https://github.com/chris48s/v8r/issues/159

**Comment by chris48s** [comment]
I've just released [v0.13.1](https://www.npmjs.com/package/v8r/v/0.13.1) with a fix for this.

**Comment by ryanwalder** [comment]
Just had a chance to get back to this. Can confirm it's fixed for me. Thanks!
